File: week-04/day-02/Aircraft/Carrier.py
from Aircraft import Aircraft

class Carrier():
    def __init__(self, store_of_ammo, healt_point):
        self.list_of_aircraft = []
        self.store_of_ammo = store_of_ammo
        self.healt_point = healt_point


    def addAircraft(self, aircraft):
        self.list_of_aircraft.append(aircraft)


    def fill(self):
        for item in self.list_of_aircraft:
            if item.getType() == "F35":
                self.store_of_ammo = item.refill(self.store_of_ammo)
        for item in self.list_of_aircraft:
            if item.getType() == "F16":
                self.store_of_ammo = item.refill(self.store_of_ammo)
        
    def fight(self, Carrier):
        carrier_max_dmg = 0
        for item in self.list_of_aircraft:
            carrier_max_dmg += item.fight()
        Carrier.healt_point -= carrier_max_dmg 
        if Carrier.healt_point >= 0:
            self.getStatus()
        else:
            print("Dead!")    

    def getStatus(self):
        carrier_max_dmg = 0
        for item in self.list_of_aircraft:
            carrier_max_dmg += item.fight()
        print("HP: {}, Aircrfat count: {}, Ammo Storage: {}, Total damage: {}".format(self.healt_point, len(self.list_of_aircraft), self.store_of_ammo, carrier_max_dmg ))
        for item in self.list_of_aircraft:
            item.getStatus()
        

carrier = Carrier(3000,5000)

aircraft01 = Aircraft("F35")
carrier.addAircraft(aircraft01)
aircraft02 = Aircraft("F16")
carrier.addAircraft(aircraft02)
aircraft03 = Aircraft("F35")
carrier.addAircraft(aircraft03)
aircraft04 = Aircraft("F16")
carrier.addAircraft(aircraft04)
aircraft05 = Aircraft("F35")
carrier.addAircraft(aircraft05)
aircraft06 = Aircraft("F16")
carrier.addAircraft(aircraft06)
aircraft07 = Aircraft("F35")
carrier.addAircraft(aircraft07)
aircraft08 = Aircraft("F35")
carrier.addAircraft(aircraft08)
aircraft09 = Aircraft("F16")
carrier.addAircraft(aircraft09)
aircraft10 = Aircraft("F35")
carrier.addAircraft(aircraft10)
aircraft11 = Aircraft("F16")
carrier.addAircraft(aircraft11)
aircraft12 = Aircraft("F35")
carrier.addAircraft(aircraft12)

carrier.getStatus()
carrier.fill()
carrier.getStatus()

File: Csharp-course/week-03/day-05/AircraftCarrier/AircraftCarrier/Carrier.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AircraftCarrier
{
    class Carrier
    {
        protected List<Aircraft> listOfAircraft = new List<Aircraft>();
        protected int storeOfAmmo;
        protected int healthPoint;

        public Carrier(int ammo, int health)
        {
            this.listOfAircraft.Clear();
            this.storeOfAmmo = ammo;
            this.healthPoint = health;
        }

        public void Add(Aircraft newAircraft)
        {
            this.listOfAircraft.Add(newAircraft);
        }

        public void fill(int ammo)
        {
            if (ammo <= this.storeOfAmmo)
            {
                foreach (Aircraft actualAircraft in this.listOfAircraft)
                {
                    if (actualAircraft.isPriority() && actualAircraft.currentAmmo < actualAircraft.maxAmmo)
                    {
                        this.storeOfAmmo -= actualAircraft.Refill(ammo);
                        ammo -= actualAircraft.Refill(ammo);
                    }
                }
                foreach (Aircraft actualAircraft in this.listOfAircraft)
                {
                    if (actualAircraft.isPriority() == false && actualAircraft.currentAmmo < actualAircraft.maxAmmo)
                    {
                        this.storeOfAmmo -= actualAircraft.Refill(ammo);
                        ammo -= actualAircraft.Refill(ammo);
                    }
                }
            }
        }

        public void Fight(Carrier enemyCarrier)
        {
            enemyCarrier.healthPoint -= this.MaxDamage();
            foreach (Aircraft actualAircraft in this.listOfAircraft)
            {
                actualAircraft.currentAmmo = 0;
            }
            if (enemyCarrier.healthPoint <= 0)
            {
                enemyCarrier.healthPoint = 0;
                Console.WriteLine("It's dead Jim :(");
            }
        }

        public int MaxDamage()
        {
            int maxDamage = 0;
            foreach (Aircraft actualAircraft in this.listOfAircraft)
            {
                maxDamage += actualAircraft.baseDamage * actualAircraft.currentAmmo;
            }
            return maxDamage;
        }

        public void getStatus()
        {
            Console.WriteLine("HP: " + this.healthPoint + ", Aircraft count: "+this.listOfAircraft.Count+
                ", Ammo Storage: "+this.storeOfAmmo+", Total Damage: "+ this.MaxDamage());
            Console.WriteLine("Aircrafts:");
            foreach (Aircraft actualAircraft in this.listOfAircraft)
            {
                Console.WriteLine("Type: {0}, Ammo: {1}, Base Damage: {2}, All Damage: {3}",actualAircraft.getType(), actualAircraft.currentAmmo, actualAircraft.baseDamage, actualAircraft.baseDamage * actualAircraft.currentAmmo);
            }
        }
    }
}
